src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.spec.ts

```diff
@@ -47,10 +51,131 @@ describe('InstructorCourseEditPageComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorCourseEditPageComponent);
     component = fixture.componentInstance;
+    courseService =  TestBed.inject(CourseService);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should load correct course details for given API output', () => {
+    spyOn(courseService, 'getCourseAsInstructor').and.returnValue(of({
+      courseId: 'exampleId',
+      courseName: 'Example Course',
+      timeZone: 'UTC (UTC)',
+      creationTimestamp: 0,
+      deletionTimestamp: 1000,
+    }));
+
+    component.loadCourseInfo();
+
+    expect(component.course.courseId).toBe('exampleId');
+    expect(component.course.courseName).toBe('Example Course');
+    expect(component.course.timeZone).toBe('UTC (UTC)');
+    expect(component.course.creationTimestamp).toBe(0);
+    expect(component.course.deletionTimestamp).toBe(1000);
+    expect(component.hasCourseLoadingFailed).toBeFalsy();
+  });
+
+  it('should not change course details if CANCEL is requested', () => {
+    spyOn(courseService, 'getCourseAsInstructor').and.returnValue(of({
+      courseId: 'exampleId',
+      courseName: 'Example Course',
+      timeZone: 'UTC (UTC)',
+      creationTimestamp: 0,
+      deletionTimestamp: 1000,
+    }));
+    component.loadCourseInfo();
```

Comment:
I think you can set `component.course` attribute directly here to initialise data, since you are not testing courseService API again.